tests/vm_nesting.sh

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
+
+function parallel() {
+        seq 1 $1 | xargs -P $1 -I "{}" "${@:2}"
+}
+
+function init() {
+	parallel $1 lxc init ubuntu:jammy "t{}" $2
+}
+
+function conf() {
+	parallel $1 lxc config set "t{}" $2
+}
+
+function device() {
+	parallel $1 lxc config device add "t{}" $2 $3 $4
+}
+
+function start() {
+	args=""
+
+	for i in $(seq 1 $1); do
+		args="t$i $args"
+	done
+
+	echo "start $args"
+	lxc start $args
+}
+
+function wait() {
+	parallel $1 bash -c "while true; do if lxc shell t{}; then break; fi; sleep 1; done"
+}
+
+function copy() {
+	parallel $1 lxc file push $2 "t{}$3"
+}
+
+function cmd() {
+	parallel $1 lxc exec "t{}" -- bash -c "$2"
+}
+
+function delete() {
+	args=""
+
+	for i in $(seq 1 $1); do
+                args="t$i $args"
+        done
+
+	echo "delete $args"
+	lxc delete -f $args
+}
+
+if [[ ! -f $1 ]]
+then
+	echo "Specify the path to the LXD binary for sideloading"
+	exit 1
+fi
+
+# Test 10 VMs in parallel
+init 10 --vm
+start 10
+delete 10
+
+# Test vsock ID collision
+init 10 --vm
+conf 10 volatile.vsock_id=42
+start 10
+delete 10
+
+# Test 5 VMs each with one nested VM
+init 5 --vm
+start 5
+wait 5
+cmd 5 "snap wait system seed.loaded && snap refresh lxd --channel latest/stable"
```

Comment:
Can these tests always use latest/edge so we can test pre-release versions too

Reply:
I have added a variable to select the edge channel. If `lxc` isn't already present on the host the test runs on, it gets installed. I haven't added a `snap refresh` so that the test doesn't modify our test environments if executed on the dev machines.

Reply:
You should be running latest/edge on dev machines though ;)

Reply:
I have the bleeding edge always sideloaded :)

Reply:
Yes, but you need to also be running the latest/edge channel so that your sideloaded binary is being run in the latest snap environment and with latest dependencies (such as qemu):

https://github.com/canonical/lxd-pkg-snap/commits/latest-edge

Reply:
Ah yes, thanks for the hint! Just changed it.